### ISSUE TITLE
MINUIT output not shown in Jupyter notebook

### ISSUE THREAD

**Issue by cdeil** [scikit-hep/iminuit#295]
When fitting with `print_level=2` from Jupyter, the extra output goes into the terminal where Jupyter was started, and doesn't appear in the notebook output cell.

@HDembinski or @piti118 or anyone - is it possible to handle that extra output so that it appears together with other text output in the Jupyter notebook, where users expect to see it?

**Comment by HDembinski** [comment]
The extra output is generated by MINUIT2. The stuff that is normally shown is generated by iminuit.

**Comment by HDembinski** [comment]
There is no way to fix this with the current setup as far as I can see.

**Comment by cdeil** [comment]
@HDembinski - So MINUIT2 just writes to stderr or stdout? Do you think it's a good idea to make a feature request to make the configurable, so that wrappers like iminuit can do a better job? Or is it not a good idea, because too much hassle on the C++ and / or Python wrapper side?

I still wonder if a Python / Numpy rewrite of MINUIT wouldn't be the way to go overall, e.g. proposed as a Google summer of code project. Most other modern projects (e.g. pytorch) re-implement the optimiser algorithms from scratch and it's usually 10 or 20 lines. I know for MINUIT it will be a few 100 lines and very difficult to get consistent, but with ~ 3 months of work like a GSoC project it should be doable, no? Having a `minuit.py` that just imports Numpy and is otherwise pure Python would be so nice...

**Comment by cdeil** [comment]
Another idea (besides just closing this issue) could be to introduce a small page called FAQ or Notes with such notes like explaining the gotcha that when using Jupyter some output will go to the output cell and some to the terminal where Jupyter was started.

**Comment by HDembinski** [comment]
I would love to see the MnPrint facility in MINUIT C++ code fixed - which is responsible for debug and info output, but it is a complete mess. I doubt that a fix is not going to happen, because it is a lot of work and the ROOT devs are busy with other things. I already noticed this problem a while ago when I noticed buggy behavior in MINUIT and wanted to increase the print level to see what is happening.
